### PR TITLE
Animation Cleanup: En_G*

### PR DESCRIPTION
**Pull request by engineer124**

### CODE REVIEW

**Review by AngheloAlf — APPROVED** [review]

File: src/overlays/actors/ovl_En_Go/z_en_go.c [hunk 1]
```diff
@@ -95,20 +95,20 @@ typedef enum EnGoAnimationIndex {
...
+    /* 14 */ ENGO_ANIM_SITTING_STRETCH,
+    /* 15 */ ENGO_ANIM_CHEER,
+    /* 16 */ ENGO_ANIM_SHOUT,
+    /* 17 */ ENGO_ANIM_HELP_SITTING_STRETCH,
+    /* 18 */ ENGO_ANIM_SPRING_MIN,
+    /* 18 */ ENGO_ANIM_SHOW = ENGO_ANIM_SPRING_MIN,
+    /* 19 */ ENGO_ANIM_SHOW_LOOPED,
+    /* 20 */ ENGO_ANIM_LOOK_AROUND,
+    /* 21 */ ENGO_ANIM_LOOK_AROUND_LOOPED
 } EnGoAnimationIndex;
```
Comment (hensldm) [inline src/overlays/actors/ovl_En_Go/z_en_go.c:112]:
```suggestion
    /* 21 */ ENGO_ANIM_LOOK_AROUND_LOOPED,
    /* 22 */ ENGO_ANIM_MAX
} EnGoAnimation;
```

File: src/overlays/actors/ovl_En_Gm/z_en_gm.c [hunk 2]
```diff
@@ -209,20 +208,38 @@ static ColliderSphereInit sSphereInit = {
...
-    { &object_in2_Anim_00898C, 1.0f, 0, -1, ANIMMODE_ONCE, -4 },
-    { &object_in2_Anim_009450, 1.0f, 0, -1, ANIMMODE_ONCE, -4 },
-    { &object_in2_Anim_00AD18, 1.0f, 0, -1, ANIMMODE_LOOP, 0 },
-    { &object_in2_Anim_00AD18, 1.0f, 0, -1, ANIMMODE_LOOP, -4 },
-    { &object_in2_Anim_00B8B0, 1.0f, 0, -1, ANIMMODE_ONCE, 0 },
-    { &object_in2_Anim_00BA80, 1.0f, 0, -1, ANIMMODE_LOOP, -4 },
-    { &object_in2_Anim_00C03C, 1.0f, 0, -1, ANIMMODE_LOOP, -4 },
-    { &object_in2_Anim_00B990, 1.0f, 0, -1, ANIMMODE_LOOP, -4 },
+typedef enum {
+    /*   -1 */ ENGM_ANIM_NONE = -1,
```
Comment (hensldm) [inline src/overlays/actors/ovl_En_Gm/z_en_gm.c:212]:
```suggestion
    /*  -1 */ ENGM_ANIM_NONE = -1,
```

File: src/overlays/actors/ovl_En_Go/z_en_go.c [hunk 3]
```diff
@@ -1701,7 +1701,7 @@ void EnGo_ChangeToSpectatingAnimation(EnGo* this, PlayState* play) {
 void EnGo_ChangeToFrozenAnimation(EnGo* this, PlayState* play) {
     Collider_InitAndSetCylinder(play, &this->colliderCylinder, &this->actor, &sCylinderInitFrozen);
     CollisionCheck_SetInfo2(&this->actor.colChkInfo, &sDamageTable, &sColChkInfoInit);
-    this->curAnimIndex = -1;
+    this->animIndex = -1;
```
Comment (hensldm) [inline src/overlays/actors/ovl_En_Go/z_en_go.c:1704]: Should this use the invalid enum value? Should the enum actually be `NONE`?

File: src/overlays/actors/ovl_En_Gg/z_en_gg.c [hunk 4]
```diff
@@ -275,47 +303,47 @@ void func_80B35634(EnGg* this, PlayState* play) {
...
 
                 case 4:
-                    do {
-                        this->unk_344.unk_30 = this->unk_2DA = 11;
-                    } while (0);
-                    this->unk_2E6 = 11;
+                    this->unk_344.animIndex = this->csAnimIndex = ENGG_ANIM_11;
+                    //! FAKE:
+                    if (1) {}
+                    this->animIndex = ENGG_ANIM_11;
```
[the remark this answers is not in the document]
Comment (hensldm) [reply src/overlays/actors/ovl_En_Gg/z_en_gg.c:324]: fwiw seems to only be needed in cases 5 and 6

File: src/overlays/actors/ovl_En_Ginko_Man/z_en_ginko_man.c [hunk 5]
```diff
@@ -46,16 +45,17 @@ typedef enum {
...
-    { &object_boj_Anim_0043F0, 1.0f, 0.0f, 0.0f, ANIMMODE_LOOP, -4.0f },
-    { &object_boj_Anim_004F40, 1.0f, 0.0f, 0.0f, ANIMMODE_ONCE, -4.0f },
-    { &object_boj_Anim_000AC4, 1.0f, 0.0f, 0.0f, ANIMMODE_LOOP, -4.0f }, // looking around for customers
-    { &object_boj_Anim_004A7C, 1.0f, 0.0f, 0.0f, ANIMMODE_LOOP, -4.0f },
+static AnimationInfo sAnimationInfo[GINKO_ANIM_MAX] = {
+    { &object_boj_Anim_0008C0, 1.0f, 0.0f, 0.0f, ANIMMODE_LOOP, -4.0f }, // GINKO_ANIM_LEGSMACKING
+    { &object_boj_Anim_0043F0, 1.0f, 0.0f, 0.0f, ANIMMODE_LOOP, -4.0f }, // GINKO_ANIM_SITTING
+    { &object_boj_Anim_004F40, 1.0f, 0.0f, 0.0f, ANIMMODE_ONCE, -4.0f }, // GINKO_ANIM_REACHING
+    { &object_boj_Anim_000AC4, 1.0f, 0.0f, 0.0f, ANIMMODE_LOOP, -4.0f }, // GINKO_ANIM_AMAZED
+    { &object_boj_Anim_004A7C, 1.0f, 0.0f, 0.0f, ANIMMODE_LOOP, -4.0f }, // GINKO_ANIM_ADVERTISING
```
[the remark this answers is not in the document]
Comment (engineer124) [reply src/overlays/actors/ovl_En_Ginko_Man/z_en_ginko_man.c]: Oh, nice catch. I just moved the comment to advertising but I'll leave "amazed" for a later doc pass? I could add a comment though, any suggestions?

**Review by jpburnett — APPROVED** [review]
Looks En_Good to me

File: src/overlays/actors/ovl_En_Ginko_Man/z_en_ginko_man.c (hunk 5 above)
[the remark this answers is not in the document]
Comment (isghj5) [reply src/overlays/actors/ovl_En_Ginko_Man/z_en_ginko_man.c]: Something that specifies that its used when the player is near but not yet speaking with the banker, like

`// used when player close/approaching banker, before dialogue started` ?

**Review by isghj5 — APPROVED** [review]
The rest looks fine, no need to wait for the extra comment.

**Review by tom-overton — APPROVED** [review]

**Pull request merged**